src/main/java/hudson/plugins/cobertura/CoberturaPublisher.java

```diff
@@ -306,46 +308,34 @@ public void setFailingTarget(CoverageTarget failingTarget) {
      * Gets the directory where the Cobertura Report is stored for the given project.
      */
     /*package*/
-    static File getCoberturaReportDir(AbstractItem project) {
-        return new File(project.getRootDir(), "cobertura");
-    }
-
-    /**
-     * Gets the directory where the Cobertura Report is stored for the given project.
-     */
-    /*package*/
-    static File[] getCoberturaReports(AbstractBuild<?, ?> build) {
+    static File[] getCoberturaReports(Run<?, ?> build) {
         return build.getRootDir().listFiles(COBERTURA_FILENAME_FILTER);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) 
+    public void perform(Run<?, ?> build, FilePath workspace, Launcher launcher, final TaskListener listener)
             throws InterruptedException, IOException {
         Result threshold = onlyStable ? Result.SUCCESS : Result.UNSTABLE;
         if (build.getResult().isWorseThan(threshold)) {
```

Comment:
This line needs to have a check for if build.getResult() returns null, as during the pipeline (which is when this will be run) there is no build result unless one is explicitly set. In my opinion a null build result should be treated as if the build failed, possibly with a seperate message stating that a build result needs to be set prior to this running.